jobs/transforms/auth_audits.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Beam DoFns and PTransforms to provide validation of auth models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import feconf
+from jobs.decorators import audit_decorators
+from jobs.transforms import base_model_audits
+
+(auth_models, user_models) = (
+    models.Registry.import_models([models.NAMES.auth, models.NAMES.user]))
+
+
+@audit_decorators.AuditsExisting(auth_models.FirebaseSeedModel)
+class ValidateFirebaseSeedModelId(base_model_audits.ValidateBaseModelId):
+    """Overrides regex to match the single valid FirebaseSeedModel ID."""
+
+    def __init__(self):
+        super(ValidateFirebaseSeedModelId, self).__init__()
+        self._pattern = auth_models.ONLY_FIREBASE_SEED_MODEL_ID
+
+
+@audit_decorators.AuditsExisting(auth_models.UserIdByFirebaseAuthIdModel)
+class ValidateUserIdByFirebaseAuthIdModelId(
+        base_model_audits.ValidateBaseModelId):
+    """Overrides regex to match the Firebase account ID pattern."""
+
+    def __init__(self):
+        super(ValidateUserIdByFirebaseAuthIdModelId, self).__init__()
+        self._pattern = feconf.FIREBASE_AUTH_ID_REGEX
+
+
+@audit_decorators.RelationshipsOf(auth_models.UserAuthDetailsModel)
```

Comment:
This is quite cool :)

One thought though. I wonder if we should put this information in the storage layer, in the model class itself? Perhaps ditto for anything model-specific. Would that be better? That way, all the model stuff is in one place rather than spread out across the codebase (I don't think someone adding a new field to a model will think to check here, for instance).

Reply:
Yeah, after refactoring things to use the new ModelProperty object, it seems like it'd be better to introduce an `IdProperty` subclass of `Property` that carries this information inside of it. I don't want to block the PR on this, but it might be worth looking into.

Reply:
Sg. I don't mind how it's done (IdProperty seems sensible) but please track it; it would be good to have all the model stuff in one self-contained place. Otherwise people are going to forget to update this part.

Speaking of which ... in the interim state, what defence do you have against people forgetting to update this part? Any backend tests to validate?

Reply:
I don't have any preference, especially considering the vast majority of validators are being tracked already in our spreadsheet. The goal here is to make it easy to add new ones, let's worry about enforcement in separate PRs (although I don't think it'll be easy without tagging properties somehow)

Reply:
OK, please track, but otherwise sg.

Reply:
Done (in idea doc).